aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLOperation.java

```diff
@@ -142,6 +142,9 @@ public void onResponse(@NonNull Call call, @NonNull Response response) {
                     return;
                 }
             }
+            if(response.code() >= 400 && response.code() <= 499) {
+                onFailure.accept(new ApiException.NonRetryableException("OkHttp client request failed.", "Irrecoverable error"));
+            }
```

Comment:
Probably need to add this to `MultiAuthAppSyncGraphQLOperation`. Looking at that code, I think it would probably have to be right before [this else statement](https://github.com/aws-amplify/amplify-android/blob/05715041669c97d4e39f3c5c1d75045e771f151e/aws-api/src/main/java/com/amplifyframework/api/aws/MultiAuthAppSyncGraphQLOperation.java#L182-L184)

Reply:
Actually, we may even want to do it as a follow up PR given the added complexity of the multiauth logic.

Reply:
Created a task for it.